### PR TITLE
Fix NRE for __arglist

### LINKED ISSUE #5965: `__arglist` parameter causes CA2016 to crash
### Analyzer

**Diagnostic ID**: [CA2016](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2016): `Forward the CancellationToken parameter to methods that take one`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.100](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

```
Analyzer 'Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpForwardCancellationTokenToInvocationsAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
Exception occurred with following context:
Compilation: ICSharpCode.Decompiler.Tests
IOperation: Invocation
SyntaxTree: \ILSpy\ICSharpCode.Decompiler.Tests\TestCases\Correctness\UndocumentedExpressions.cs
SyntaxNode: VarArgs(1, __arglist()) [InvocationExpressionSyntax]@[1343..1366) (27,3)-(27,26)

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.NetCore.Analyzers.Runtime.ForwardCancellationTokenToInvocationsAnalyzer.<>c__DisplayClass12_0.<ShouldDiagnose>b__0(IArgumentOperation a)
   at Microsoft.NetCore.Analyzers.Runtime.ForwardCancellationTokenToInvocationsAnalyzer.AnyArgument(ImmutableArray`1 arguments, Func`2 predicate)
   at Microsoft.NetCore.Analyzers.Runtime.ForwardCancellationTokenToInvocationsAnalyzer.ShouldDiagnose(Compilation compilation, IInvocationOperation invocation, IMethodSymbol containingSymbol, INamedTypeSymbol cancellationTokenType, INamedTypeSymbol genericTask, INamedTypeSymbol genericValueTask, Int32& shouldFix, String& ancestorTokenParameterName, String& invocationTokenParameterName)
   at Microsoft.NetCore.Analyzers.Runtime.ForwardCancellationTokenToInvocationsAnalyzer.<>c__DisplayClass11_0.<AnalyzeCompilationStart>b__0(OperationAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteOperationAction>b__63_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```

https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs#L138 causes the bug.

Could it be that `__arglist` in `public static void VarArgs(int normalArg, __arglist)` does not have a type?

### Steps To Reproduce

1. Get ILSpy's source code.
2. Follow https://github.com/icsharpcode/ILSpy#how-to-build
3. Open any of the ILSpy solution (filters)
4. Go to ICSharpCode.Decompiler.Tests\TestCases\Correctness\UndocumentedExpressions.cs
5. Wait for the analyzer to be triggered

### Expected behavior

No crash.

### Actual behavior

Crash.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #5965

I originally had this as part of the blocked PR #5865.

I'll revise #5865 to see if there are any other bugs it revealed and fix them separately.

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani @carlossanlop To review a quick fix.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs [hunk 1]
```diff
@@ -135,7 +135,7 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
 
             // Verify that the current invocation is not passing an explicit token already
             if (AnyArgument(invocation.Arguments,
-                            a => a.Parameter.Type.Equals(cancellationTokenType) && !a.IsImplicit))
+                            a => cancellationTokenType.Equals(a.Parameter?.Type) && !a.IsImplicit))
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs:138]: @Youssef1313 Is the generated IOperation tree nullable-aware?

**Review by mavasani — APPROVED** [review]
Thanks!

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs (hunk 1 above)
Comment (Youssef1313) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs:138]: @mavasani Yes, but for a newer version of Microsoft.CodeAnalysis than what we use here.

That's why #5865 has revealed this bug.

Comment (Youssef1313) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs:138]: For reference, `IArgumentOperation.Parameter` was annotated as nullable in https://github.com/dotnet/roslyn/commit/456829a7f271db03faa2889e2f959b60d8c2ce07.

**Pull request merged**